src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoProjectDecorator.java

```diff
@@ -1,300 +0,0 @@
-        delegate.createService(project, serviceTemplate, serviceCRD, service, spec, wait);
-    }
-
-    @Override
-    public String getServiceTemplate(String project, String service) throws IOException {
-        return delegate.getServiceTemplate(project, service);
-    }
-
-    @Override
-    public void deleteService(String project, Service service) throws IOException {
-        delegate.deleteService(project, service);
-    }
-
-    @Override
-    public List<DevfileComponentType> getComponentTypes() throws IOException {
-        return delegate.getComponentTypes();
-    }
-
-    @Override
-    public List<ServiceTemplate> getServiceTemplates() throws IOException {
-        return delegate.getServiceTemplates();
-    }
-
-    @Override
-    public void describeServiceTemplate(String template) throws IOException {
-        delegate.describeServiceTemplate(template);
-    }
-
-    @Override
-    public List<URL> listURLs(String project, String context, String component) throws IOException {
-        return delegate.listURLs(project, context, component);
-    }
-
-    @Override
-    public ComponentInfo getComponentInfo(String project, String component, String path,
-                                          ComponentKind kind) throws IOException {
-        return delegate.getComponentInfo(project, component, path, kind);
-    }
-
-    @Override
-    public void deleteComponent(String project, String context, String component,
-                                ComponentKind kind) throws IOException {
-        if (context != null) {
-            root.removeContext(new File(context));
-        }
-        delegate.deleteComponent(project, context, component, kind);
-    }
-
-    @Override
-    public void follow(String project, String context, String component, boolean deploy, String platform) throws IOException {
-        delegate.follow(project, context, component, deploy, platform);
-    }
-
-    @Override
-    public void log(String project, String context, String component, boolean deploy, String platform) throws IOException {
-        delegate.log(project, context, component, deploy, platform);
-    }
-
-    @Override
-    public boolean isLogRunning(String context, String component, boolean deploy) throws IOException {
-        return delegate.isLogRunning(context, component, deploy);
-    }
-
-    @Override
-    public void createProject(String project) throws IOException {
-        delegate.createProject(project);
-    }
-
-    @Override
-    public void deleteProject(String project) throws IOException {
-        delegate.deleteProject(project);
-    }
-
-    @Override
-    public void login(String url, String userName, char[] password, char[] token) throws IOException {
-        delegate.login(url, userName, password, token);
-    }
-
-    @Override
-    public List<Component> getComponents(String project) throws IOException {
-        List<Component> components = delegate.getComponents(project);
```

Comment:
please move this logic i in the calling action

Reply:
I believe that I cannot remove `OdoProjectDecorator`. Turns out that this "decorator" does things to the tree before/after invoking the odo binary. Example:
```java
    public void deleteComponent(String project, String context, String component, ComponentKind kind) throws IOException {
        if (context != null) {
            root.removeContext(new File(context));
        }
        delegate.deleteComponent(project, context, component, kind);
    }
```
I could implement the `Odo` interface in `ApplicationsRootNode` but would loose the separation of concerns that we have now. 
To me it looks as if keeping the decorator is the better option. But I'll rename it to `ApplicationsRootNodeOdo` to better reflect its purpose.